### PR TITLE
Fixes #33091 - Trigger recurring logic with a delay

### PR DESCRIPTION
**Pull request by xprazak2**
I plan to pass [start_at](https://github.com/theforeman/foreman_remote_execution/pull/619/files#diff-c7a804360df9b6b127e28adfdaa63daa5fcd5af31d82886dab5e4ff4d3708baaR150) on rex side.

### CODE REVIEW

**Comment by adamruzicka** [comment]
Looks fine on first glance, will wait for the rex part

**Comment by xprazak2** [comment]
The rex part: https://github.com/theforeman/foreman_remote_execution/pull/627

File: app/models/foreman_tasks/triggering.rb [hunk 1]
```diff
@@ -104,6 +104,7 @@ def parse_start_before!
     private
 
     def can_start_recurring
+      parse_start_at!
```
Comment (adamruzicka) [inline app/models/foreman_tasks/triggering.rb:107]: Could we add a validation message instead of just raising?

**Review by adamruzicka — COMMENTED** [review]
One small nitpick.

@jlsherrill Could you check if this doesn't break some assumption katello makes when using sync plans? I don't expect really it, but better to be safe

File: app/models/foreman_tasks/triggering.rb (hunk 1 above)
Comment (xprazak2) [reply app/models/foreman_tasks/triggering.rb:107]: I do not think [parse_start_at!](https://github.com/theforeman/foreman-tasks/blob/master/app/models/foreman_tasks/triggering.rb#L96) raises:

```ruby
Time.zone.parse('foo') #=> nil
```

Comment (adamruzicka) [reply app/models/foreman_tasks/triggering.rb:107]: It does ocasionally

```
> Time.zone.parse('2021-08-32')
ArgumentError: argument out of range
from /home/aruzicka/.rvm/gems/ruby-2.7.0/gems/activesupport-6.0.3.7/lib/active_support/values/time_zone.rb:548:in `initialize'

> Time.zone.parse('127.0.0.1')
ArgumentError: mon out of range
from /home/aruzicka/.rvm/gems/ruby-2.7.0/gems/activesupport-6.0.3.7/lib/active_support/values/time_zone.rb:548:in `initialize'
```

Comment (xprazak2) [reply app/models/foreman_tasks/triggering.rb]: Updated.

**Comment by adamruzicka** [comment]
Just to double check if I'm reading this right. If I schedule a job to start next monday at noon and run daily, then the first run will happen on monday at noon, next run at midnight between monday and tuesday and then continue happening at midnights?

**Comment by xprazak2** [comment]
The schedule is always honored. If I schedule a job to run daily at midnight starting next monday at noon, then the first run will be at midnight between monday and tuesday.

run-schedule

run-detail

**Comment by xprazak2** [comment]
Anything I can do to move this forward?

**Review by adamruzicka — APPROVED** [review]
Works well, ACK

**Comment by adamruzicka** [comment]
Thank you @xprazak2 !

**Pull request merged**